From e6975ad2f2b233d698f78c19546a6e65019974e6 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 19 Dec 2019 23:39:47 +0200
Subject: [PATCH] [SW-0] spi: fix the name printing

The spi_copy() function receives a name string of the component it works
on, so it can print it to the console. This way we can differentiate
between the different spi_copy() function invocations.
While printing the name, it calls preboot_puts() function which does not
output the trailing EOL ('\n') character, but it does output the
trailing CR ('\r') character, so the next call to preboot_puts()
overwrites the actual name string.
Fix this by switching the preboot_puts() call to preboot_puts_raw() call
which does not output the trailing CR character.

Change-Id: If59bbd5cb8461cd1b6384b37d923ffbae026b006
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/spi.c b/common/spi.c
index 6d2d94d..c78380c 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -212,7 +212,7 @@
 	spi_init();
 	spi_read(dst, offset, size);
 	sr = spi_prot();
-	preboot_puts(name);
+	preboot_puts_raw(name);
 	preboot_puts(" copying from flash is completed\n");
 	if ((sr & PROT_MASK) == PROT_MASK)
 		preboot_puts("Flash may be protected, depend on WP# signal\n");
